src/main/java/org/opensearch/knn/index/KNNVectorScriptDocValues.java

```diff
@@ -14,16 +15,23 @@
 
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
+import java.util.Locale;
+
+import static org.opensearch.knn.common.KNNConstants.VECTOR_DATA_TYPE_FIELD;
+import static org.opensearch.knn.index.VectorDataType.SUPPORTED_VECTOR_DATA_TYPES;
 
 public final class KNNVectorScriptDocValues extends ScriptDocValues<float[]> {
 
     private final BinaryDocValues binaryDocValues;
     private final String fieldName;
+    @Getter
+    private final VectorDataType vectorDataType;
     private boolean docExists;
 
-    public KNNVectorScriptDocValues(BinaryDocValues binaryDocValues, String fieldName) {
+    public KNNVectorScriptDocValues(BinaryDocValues binaryDocValues, String fieldName, VectorDataType vectorDataType) {
```

Comment:
Is it possible to switch to lombok's AllArgsConstructor here? I think if we initialize `docExists` to `false` then the rest of fields will form same constructor as this one that defined explicitly

Reply:
Updated using `RequiredArgsConstructor`